src/lib/libast/path/pathaccess.c

```diff
@@ -38,19 +39,21 @@
 
 #include "ast.h"
 
-char *pathaccess(const char *dirs, const char *a, const char *b, int mode, char *path,
+char *pathaccess(const char *dirs, const char *path1, const char *path2, int mode, char *path,
                  size_t size) {
-    int sib = a && a[0] == '.' && a[1] == '.' && a[2] == 0;
-    int sep = ':';
+    int is_path1_parentdir = path1 && path1[0] == '.' && path1[1] == '.' && path1[2] == 0;
```

Comment:
FYI, I am not a fan of the micro optimizations such a this that exist in the code. They often result in the code accessing bytes past the end of a buffer/string. They also are harder to read. So while doing these types of cleanup I recommend changing them to a more idiomatic formulation. For example:

```
bool is_path1_parentdir = path1 && strcmp(path1, "..") == 0;
```